### PR TITLE
suite-nonsplit-checkout

### PR DESCRIPTION
**Pull request by himanshu-saxena-metacube**
https://spryker.atlassian.net/browse/CC-16788 - Checkout

### CODE REVIEW

File: tests/api/suite_nonsplit/glue/checkout_endpoints/checkout/positive.robot [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+    And Response body parameter should contain:    [included][0][attributes][calculatedDiscounts]    sumAmount
+    And Response body parameter should contain:    [included][0][attributes][calculatedDiscounts]    displayName
+    And Response body parameter should contain:    [included][0][attributes][calculatedDiscounts]    description
+    And Response body parameter should contain:    [included][0][attributes][calculatedDiscounts]    voucherCode
+    And Response body parameter should contain:    [included][0][attributes][calculatedDiscounts]    quantity
+
+Create_order_with_net_mode_&_chf_currency_&_express_shipment_method
+    [Setup]    Run Keywords    I get access token for the customer:    ${yves_second_user.email}
+     ...  AND    I set Headers:    Authorization=${token}
+    ...  AND    I send a POST request:    /carts    {"data":{"type":"carts","attributes":{"priceMode":"${mode.gross}","currency":"${currency.eur.code}","store":"${store.de}","name": "${test_cart_name}-${random}"}}}
```
Comment (Katerynaa-Lavryk) [inline tests/api/suite_nonsplit/glue/checkout_endpoints/checkout/positive.robot:179]: According to the test title netMode and CHF currency should be checked but in fact you are checking grossMode and EUR currency. Please update the test

Comment (himanshu-saxena-metacube) [reply tests/api/suite_nonsplit/glue/checkout_endpoints/checkout/positive.robot]: Fixed

File: tests/api/suite_nonsplit/glue/checkout_endpoints/checkout/positive.robot [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+Suite Setup    SuiteSetup
+Test Setup     TestSetup
+Resource    ../../../../../../resources/common/common_api.robot
+Default Tags    glue
+
+*** Test Cases ***
+ENABLER
+    TestSetup
+
+#POST requests
```
Comment (OKrivtsova) [inline tests/api/suite_nonsplit/glue/checkout_endpoints/checkout/positive.robot:11]: please add a test for placing an order with a configurable bundle

Comment (himanshu-saxena-metacube) [reply tests/api/suite_nonsplit/glue/checkout_endpoints/checkout/positive.robot:11]: fixed

File: tests/api/suite_nonsplit/glue/checkout_endpoints/checkout/positive.robot [hunk 3]
```diff
@@ -0,0 +1,407 @@
...
+    #calculatedDiscounts - "Free standard delivery" discount
+    And Response body parameter should contain:    [included][0][attributes][calculatedDiscounts]    unitAmount: None
+    And Response body parameter should contain:    [included][0][attributes][calculatedDiscounts]    sumAmount: ${discount_3.total_sum}
+    And Response body parameter should contain:    [included][0][attributes][calculatedDiscounts]    displayName: ${discount_3.name}
+    And Response body parameter should contain:    [included][0][attributes][calculatedDiscounts]    description: ${discount_3.description}
+    And Response body parameter should contain:    [included][0][attributes][calculatedDiscounts]    voucherCode: None
+    And Response body parameter should contain:    [included][0][attributes][calculatedDiscounts]    quantity: 1
+
+#bug:https://spryker.atlassian.net/browse/CC-19233
+Create_order_with_2_product_discounts
```
[the remark this answers is not in the document]
Comment (himanshu-saxena-metacube) [reply tests/api/suite_nonsplit/glue/checkout_endpoints/checkout/positive.robot:328]: fixed

[the remark this answers is not in the document]
Comment (himanshu-saxena-metacube) [reply tests/api/suite_nonsplit/glue/checkout_endpoints/checkout/positive.robot:328]: Added CC-21301 Yves: error "Checkout payment is not valid" is shown if order is fully paid with a gift card and the shipment is free for reference

File: tests/api/suite_nonsplit/glue/checkout_endpoints/checkout/positive.robot [hunk 4]
```diff
@@ -0,0 +1,857 @@
...
+    And Response body parameter should contain:    [included][0][attributes][calculatedDiscounts]    quantity: 2
+    #calculatedDiscounts - "Free standard delivery" discount
+    And Response body parameter should contain:    [included][0][attributes][calculatedDiscounts]    unitAmount: None
+    And Response body parameter should contain:    [included][0][attributes][calculatedDiscounts]    sumAmount: ${discount_3.total_sum_for_discounts_for_products_1_2}
+    And Response body parameter should contain:    [included][0][attributes][calculatedDiscounts]    displayName: ${discount_3.name}
+    And Response body parameter should contain:    [included][0][attributes][calculatedDiscounts]    description: ${discount_3.description}
+    And Response body parameter should contain:    [included][0][attributes][calculatedDiscounts]    voucherCode: None
+    And Response body parameter should contain:    [included][0][attributes][calculatedDiscounts]    quantity: 1 
+
+create_order_with_configurable_bundle_item
```
Comment (OKrivtsova) [inline tests/api/suite_nonsplit/glue/checkout_endpoints/checkout/positive.robot:402]: tests should start with a capital letter

File: tests/api/suite_nonsplit/glue/checkout_endpoints/checkout/positive.robot [hunk 5]
```diff
@@ -0,0 +1,857 @@
...
+    #calculatedDiscounts
+    And Response body parameter should contain:    [included][0][attributes][calculatedDiscounts]    unitAmount
+    And Response body parameter should contain:    [included][0][attributes][calculatedDiscounts]    sumAmount
+    And Response body parameter should contain:    [included][0][attributes][calculatedDiscounts]    displayName
+    And Response body parameter should contain:    [included][0][attributes][calculatedDiscounts]    description
+    And Response body parameter should contain:    [included][0][attributes][calculatedDiscounts]    voucherCode
+    And Response body parameter should contain:    [included][0][attributes][calculatedDiscounts]    quantity
+
+#bug https://spryker.atlassian.net/browse/CC-21301
+create_checkout_with_gift_card
```
Comment (OKrivtsova) [inline tests/api/suite_nonsplit/glue/checkout_endpoints/checkout/positive.robot:549]: tests should start with a capital letter

File: tests/api/suite_nonsplit/glue/checkout_endpoints/checkout/positive.robot [hunk 6]
```diff
@@ -0,0 +1,857 @@
...
+    #calculatedDiscounts
+    And Response body parameter should contain:    [included][0][attributes][calculatedDiscounts]    unitAmount
+    And Response body parameter should contain:    [included][0][attributes][calculatedDiscounts]    sumAmount
+    And Response body parameter should contain:    [included][0][attributes][calculatedDiscounts]    displayName
+    And Response body parameter should contain:    [included][0][attributes][calculatedDiscounts]    description
+    And Response body parameter should contain:    [included][0][attributes][calculatedDiscounts]    voucherCode
+    And Response body parameter should contain:    [included][0][attributes][calculatedDiscounts]    quantity
+    
+#bug https://spryker.atlassian.net/browse/CC-21301
+create_checkout_with_gift_card_covered_some_part
```
Comment (OKrivtsova) [inline tests/api/suite_nonsplit/glue/checkout_endpoints/checkout/positive.robot:701]: tests should start with a capital letter
what does 'covered some part' mean?
please name the test according to what it tests

File: tests/api/suite_nonsplit/glue/checkout_endpoints/checkout/positive.robot (hunk 5 above)
Comment (himanshu-saxena-metacube) [reply tests/api/suite_nonsplit/glue/checkout_endpoints/checkout/positive.robot]: Done

File: tests/api/suite_nonsplit/glue/checkout_endpoints/checkout/positive.robot (hunk 6 above)
Comment (himanshu-saxena-metacube) [reply tests/api/suite_nonsplit/glue/checkout_endpoints/checkout/positive.robot]: Done

File: tests/api/suite_nonsplit/glue/checkout_endpoints/checkout/positive.robot (hunk 4 above)
Comment (himanshu-saxena-metacube) [reply tests/api/suite_nonsplit/glue/checkout_endpoints/checkout/positive.robot]: Fixed

**Review by OKrivtsova — DISMISSED** [review]

**Comment by himanshu-saxena-metacube** [comment]
@OKrivtsova Fixed conflicts, please review it again

**Review by OKrivtsova — APPROVED** [review]

**Pull request merged**